### ISSUE TITLE
dotspacemacs-emacs-leader-key can't be nil (Was: Can't type 'm' in insert mode)

### ISSUE THREAD

**Issue by kendall** [syl20bnr/spacemacs#4204]
On latest develop and latest packages (3507bc74455a213d3fcaff408453933496ef83bc) and I can't type an 'm' in insert mode when using the asciidoc package. An 'm' in insert mode brings up the "Major modes commands' minibuffer instead.

**Comment by justbur** [comment]
I can't reproduce this.

**Comment by syl20bnr** [comment]
@kendall When was the last time you updated your develop branch ?
If you remove temporarily the asciidoc package from your config, do you still have the same bug ?

In the meantime you can revert back to the previous commit of your develop branch by examining `git reflog` output.

**Comment by kendall** [comment]
I updated develop branch to the latest, this morning.

But when I disable the 'asciidoc' package the bug goes away, so I assume it's a bug there. I updated packages this AM, too, so I think I have the latest asciidoc.

**Comment by justbur** [comment]
@kendall I've had the asciidoc layer installed since you reported this and I don't see this behavior at all. You'll have to provide more information if you expect us to reproduce it I think.

**Comment by StreakyCobra** [comment]
Same here, I'm using it for months without any problem. What would help us is:

- Pasting the output of `SPC h d s` here
- Providing a link to your `.spacemacs` if there is no sensible information in there
- Eventually a reproduction guide that starts like:
  1. Start emacs
  2. …

**Comment by kendall** [comment]
@justbur I don't expect anyone to do anything. I just reported what I've seen, given the information I have. I don't really know what is causing the bug. Disabling asciidoc package seems to have made it go away but that's all I know so far. ;>

**Comment by kendall** [comment]
#### System Info
- OS: darwin
- Emacs: 24.5.1
- Spacemacs: 0.105.0
- Spacemacs branch: develop (rev. 3507bc7)
- Distribution: spacemacs
- Layers:
```elisp
(spacemacs my-web-dev
           (colors :variables colors-enable-rainbow-identifiers t colors-enable-nyan-cat-progress-bar t)
           osx git github latex version-control markdown themes-megapack html shell extra-langs syntax-checking spell-checking emacs-lisp ibuffer vim-empty-lines vim-powerline better-defaults deft search-engine auto-completion syntax-checking
           (org :variables org-bullets-bullet-list
                '("■" "◆" "▲" "▶")
                org-enable-github-support t)
           shell-scripts javascript perspectives ruby python smex rust finance haskell erlang elixir java dash ranger clojure scala eyebrowse evil-snipe)
```

My ~/.spacemacs is https://gist.github.com/kendall/4a27d68e35d6b181a379

**Comment by justbur** [comment]
@kendall found the bug. We're not handling no emacs leader key correctly https://gist.github.com/kendall/4a27d68e35d6b181a379#file-spacemacs-el-L81

**Comment by kendall** [comment]
@justbur Thanks! I'll put something there and try to see if that 'fixes' it. I can't remember why that value is empty except that I think my spacemacs predates its introduction, perhaps.

**Comment by syl20bnr** [comment]
@kendall can you run `SPC : dotspacemacs/test-dotfile` and provide the result ?

**Comment by justbur** [comment]
@syl20bnr I'm going to fix this on the bind-map side now. I'm going to assume that  you never want to use a nil leader key, but this should probably be taken care of in spacemacs as well. 

By the way, @kendall technically it doesn't say in the documentation that you can disable the emacs leader key by setting it to nil

**Comment by kendall** [comment]
With emacs leader key set to nil:

```* Running tests on [[file:/Users/k/.spacemacs][/Users/k/.spacemacs]] (v0.0)

* Testing settings in dotspacemacs/layers [[file:/Users/k/.spacemacs::dotspacemacs/layers][Show in File]]
** TEST: Each path in [[file:/Users/k/.spacemacs::dotspacemacs-configuration-layer-path][dotspacemacs-configuration-layer-path]] is a string
** TEST: Each path in [[file:/Users/k/.spacemacs::dotspacemacs-configuration-layer-path][dotspacemacs-configuration-layer-path]] exists in filesystem
** TEST: Each layer in [[file:/Users/k/.spacemacs::dotspacemacs-configuration-layers][dotspacemacs-configuration-layers]] can be found
*** PASS: spacemacs
*** PASS: my-web-dev
*** PASS: colors
*** PASS: osx
*** PASS: git
*** PASS: github
*** PASS: latex
*** PASS: version-control
*** PASS: markdown
*** PASS: themes-megapack
*** PASS: html
*** PASS: asciidoc
*** PASS: shell
*** PASS: syntax-checking
*** PASS: spell-checking
*** PASS: emacs-lisp
*** PASS: ibuffer
*** PASS: vim-empty-lines
*** PASS: vim-powerline
*** PASS: better-defaults
*** PASS: deft
*** PASS: search-engine
*** PASS: auto-completion
*** PASS: syntax-checking
*** PASS: org
*** PASS: shell-scripts
*** FAIL: perspectives
*** PASS: smex
*** PASS: finance
*** PASS: java
*** PASS: ranger
*** PASS: eyebrowse
*** PASS: evil-snipe
** RESULTS: [[file:/Users/k/.spacemacs::dotspacemacs/layers][dotspacemacs/layers]] passed 32 out of 33 tests

* Testing settings in dotspacemacs/init [[file:/Users/k/.spacemacs::dotspacemacs/init][Show in File]]
** TEST: [[file:/Users/k/.spacemacs::dotspacemacs-editing-style][dotspacemacs-editing-style]] is 'vim, 'emacs or 'hybrid
*** PASS: vim
** TEST: [[file:/Users/k/.spacemacs::dotspacemacs-auto-save-file-location][dotspacemacs-auto-save-file-location]] is one of 'original, 'cache or nil
*** PASS: nil
** TEST: [[file:/Users/k/.spacemacs::dotspacemacs-highlight-delimiters][dotspacemacs-highlight-delimiters]] is one of 'all, 'any, 'current or nil
*** PASS: nil
** TEST: Each element of [[file:/Users/k/.spacemacs::dotspacemacs-startup-lists][dotspacemacs-startup-lists]] includes only 'recents, 'bookmarks or 'projects
** TEST: [[file:/Users/k/.spacemacs::dotspacemacs-leader-key][dotspacemacs-leader-key]] is a string
*** PASS: SPC
** TEST: [[file:/Users/k/.spacemacs::dotspacemacs-emacs-leader-key][dotspacemacs-emacs-leader-key]] is a string
*** PASS: 
** TEST: [[file:/Users/k/.spacemacs::dotspacemacs-major-mode-leader-key][dotspacemacs-major-mode-leader-key]] is a string or nil
*** PASS: ,
** TEST: [[file:/Users/k/.spacemacs::dotspacemacs-major-mode-emacs-leader-key][dotspacemacs-major-mode-emacs-leader-key]] is a string or nil
*** PASS: nil
** TEST: [[file:/Users/k/.spacemacs::dotspacemacs-command-key][dotspacemacs-command-key]] is a string
*** PASS: :
** RESULTS: [[file:/Users/k/.spacemacs::dotspacemacs/init][dotspacemacs/init]] passed 8 out of 8 tests
```

**Comment by TheBB** [comment]
He's not using nil though, it's an empty string.

**Comment by justbur** [comment]
@TheBB same thing

**Comment by justbur** [comment]
@TheBB I meant it's the same thing after passing it through `kbd`

**Comment by kendall** [comment]
The odd thing IMO is that this value has been nil for ages but I only just started seeing this issue. I guess something else must have changed to expose the issue.

Replacing nil with "SPC" seems to solve the problem.

**Comment by justbur** [comment]
@kendall "SPC" is a bad choice for the emacs leader key. Also we've been working on changing the backend for the leader keys, so it's not that odd that something would come up now.

**Comment by kendall** [comment]
@justbur what is a good choice?

**Comment by TheBB** [comment]
The default is `M-m`.

**Comment by kendall** [comment]
Thanks.

**Comment by justbur** [comment]
@syl20bnr This can't completely be fixed on the bind-map side. I'll have to submit a patch for spacemacs too.

**Comment by justbur** [comment]
Fixed in #4212

**Comment by syl20bnr** [comment]
Should be fixed in develop thanks to @justbur.
